pkg/api/v1alpha2/api.proto

```diff
@@ -0,0 +1,481 @@
+        string value = 2;
+    }
+    string name = 1;
+    repeated ValueLog valuelog = 2;
+}
+
+message Observation {
+    repeated Metric metrics = 1;
+}
+
+message ObservationLog {
+    repeated MetricLog metriclogs = 1;
+}
+
+message TrialSpec {
+    string experiment_name = 1;
+    repeated ParameterAssignment parameter_assignments = 2;
+    string run_spec = 3;
+}
+
+message TrialStatus {
+    enum TrialConditionType {
+        PENDING = 0;
+        RUNNING = 1;
+        COMPLETED = 2;
+        KILLED = 3;
+        FAILED = 4;
+    }
+    string start_time = 1; /// RFC3339 format
+    string completion_time = 2; /// RFC3339 format
+    string last_reconcile_time = 3; /// RFC3339 format
+    TrialConditionType condition = 4;
+    Observation observation = 5; /// The best observation in logs.
+}
+
+message Trial {
+    string id = 1;
+    TrialSpec spec = 2;
+    TrialStatus status = 3;
+}
+
+message RegisterExperimentRequest {
+    Experiment experiment = 1;
+}
+
+message RegisterExperimentReply {
+}
+
+message DeleteExperimentRequest {
+    string experiment_name = 1;
+}
+
+message DeleteExperimentReply {
+    string experiment_name = 1;
+}
+
+message GetExperimentRequest {
+    string experiment_name = 1;
+}
+
+message GetExperimentReply {
+    Experiment experiment = 1;
+}
+
+message GetStudyListRequest {
+}
+
+message ExperimentSummary {
+    string experiment_name = 1;
+    ExperimentStatus.ExperimentConditionType condition = 2;
+}
+
+message GetExperimentListRequest {
+}
+
+message GetExperimentListReply {
+    repeated ExperimentSummary experiment_summaries = 1;
+}
+
+message UpdateExperimentStatusRequest {
+    string experiment_name = 1;
+    ExperimentStatus new_status = 2;
+}
+
+message UpdateExperimentStatusReply {
+    string experiment_name = 1;
+}
+
+message RegisterTrialRequest {
+    Trial trial = 1;
+}
+
+message RegisterTrialReply {
+    string trial_id = 1;
+}
+
+message DeleteTrialRequest {
+    string trial_id = 1;
+}
+
+message DeleteTrialReply {
+}
+
+message GetTrialListRequest {
+    string experiment_name = 1;
+    string start_time = 2; ///The start of the time range. RFC3339 format
+    string end_time = 3; ///The end of the time range. RFC3339 format
+}
+
+message GetTrialListReply {
+    repeated Trial trials = 1;
+}
+
+message GetTrialRequest {
+    string trial_id = 1;
+}
+
+message GetTrialReply {
+    Trial trial = 1;
+}
+
+message UpdateTrialStatusRequest {
+    string trial_id = 1;
+    TrialStatus new_status = 2;
+}
+
+message UpdateTrialStatusReply {
+    string trial_id = 1;
+}
+
+message ReportObservationLogRequest {
+    string trial_id = 1;
+    ObservationLog observation_log = 2;
+}
+
+message ReportObservationLogReply {
+}
+
+message GetObservationLogRequest {
+    string trial_id = 1;
+}
+
+message GetObservationLogReply {
+    ObservationLog observation_log = 1;
+}
+
+message GetSuggestionsRequest {
+    string experiment_name = 1;
+    string algorithm_name = 2;
+    int32 request_number = 3;
```

Comment:
what is this field used for?

Reply:
When you set 3 to `request_number`, you can get three Suggestions at one time.

Reply:
Isn't this same as the `parallel_trial_count` which is already in the experiment Spec ?

Reply:
a little bit different.
When parallel_trial_count is 4 and 2 trials are completed at the same time, the operator needs to set 2 to request_number in GetSuggesitonRequest.

Reply:
Makes sense